packages/react-composites/tests/browser/call/CallComposite.test.ts

```diff
@@ -28,7 +27,59 @@ test.describe('Call Composite E2E Tests', () => {
       await pages[idx].waitForSelector(dataUiId('call-composite-local-preview'));
       await pages[idx].waitForSelector(`${dataUiId('call-composite-start-call-button')}[data-is-focusable="true"]`);
       await stubLocalCameraName(pages[idx]);
-      expect(await pages[idx].screenshot()).toMatchSnapshot(`page-${idx}-call-screen.png`);
+      expect(await pages[idx].screenshot()).toMatchSnapshot(`page-${idx}-call-screen.png`, { threshold: 0.5 });
+    }
+  });
+
+  test('local device settings can toggle camera & audio', async ({ pages }) => {
+    for (const idx in pages) {
+      const page = pages[idx];
+      page.bringToFront();
+      await stubLocalCameraName(page);
+      await page.click(dataUiId('call-composite-local-device-settings-microphone-button'));
+      await page.click(dataUiId('call-composite-local-device-settings-camera-button'));
+      await page.waitForSelector('video');
+      await page.waitForTimeout(1000);
+      expect(await page.screenshot()).toMatchSnapshot(`page-${idx}-local-device-settings-camera-enabled.png`);
+    }
+  });
+
+  test('video gallery renders for all pages', async ({ pages }) => {
+    for (const idx in pages) {
+      const page = pages[idx];
+      page.bringToFront();
+      await page.click(dataUiId('call-composite-start-call-button'));
+    }
+
+    for (const idx in pages) {
+      const page = pages[idx];
+      page.bringToFront();
+      await page.waitForFunction(() => {
+        return document.querySelectorAll('video').length === 2;
+      });
+      expect(await page.screenshot()).toMatchSnapshot(`page-${idx}-video-gallery.png`);
+    }
+  });
+
+  test('participant list loads correctly', async ({ pages }) => {
+    for (const idx in pages) {
+      const page = pages[idx];
+      page.bringToFront();
+      await page.click(dataUiId('call-composite-participants-button'));
+      await page.waitForTimeout(1000);
```

Comment:
I thought we had to wait 1s so we can compare the snapshot to the expected image but we don't always seem to need to do it for all of the tests?

Reply:
This here is a special case because clicking on participants icon displays a dropdown menu that has an animation. So it takes it a while to render completely. Hence, the wait of 1second. We don't need to wait a sec for all tests.

Reply:
ok if its a special case perhaps we should put a comment since this is specific around our UI today.